tests/devices/eiger/test_eiger.py

```diff
@@ -0,0 +1,347 @@
+from mock.mock import create_autospec
+
+# from tickit.core.device import DeviceUpdate
+# from tickit.core.typedefs import SimTime
+from tickit.devices.eiger.eiger import Eiger, EigerAdapter
+from tickit.devices.eiger.eiger_settings import EigerSettings
+from tickit.devices.eiger.eiger_status import EigerStatus, State
+
+# # # # # Eiger Tests # # # # #
+
+
+@pytest.fixture
+def eiger() -> Eiger:
+    return Eiger()
+
+
+def test_eiger_constructor():
+    Eiger()
+
+
+@pytest.mark.asyncio
+async def test_eiger_initialize(eiger):
+    await eiger.initialize()
+
+    assert eiger.get_state() == State.IDLE
+
+
+@pytest.mark.asyncio
+async def test_eiger_arm(eiger):
+    await eiger.arm()
+
+    assert eiger.get_state() == State.READY
+
+
+@pytest.mark.asyncio
+async def test_eiger_disarm(eiger):
+    await eiger.disarm()
+
+    assert eiger.get_state() == State.IDLE
+
+
+# @pytest.mark.asyncio
+# async def test_eiger_trigger(eiger):
+#     await eiger.trigger()
+
+#     assert
+
+
+@pytest.mark.asyncio
+async def test_eiger_cancel(eiger):
+    await eiger.cancel()
+
+    assert eiger.get_state() == State.READY
+
+
+@pytest.mark.asyncio
+async def test_eiger_abort(eiger):
+    await eiger.abort()
+
+    assert eiger.get_state() == State.IDLE
+
+
+def test_eiger_get_state(eiger):
+    assert eiger.get_state() == State.NA
+
+
+def test_eiger_set_state(eiger):
+
+    eiger._set_state(State.IDLE)
+
+    assert eiger.get_state() == State.IDLE
+
+
+# TODO: Tests for update() once implemented
+
+
+# # # # # EigerAdapter Tests # # # # #
+
+
+@pytest.fixture
+def mock_status() -> MagicMock:
+    return create_autospec(EigerStatus, instance=True)
+
+
+@pytest.fixture
+def mock_settings() -> MagicMock:
+    return create_autospec(EigerSettings, instance=True)
+
+
+@pytest.fixture
+def mock_eiger(mock_status, mock_settings) -> MagicMock:
+    mock_eiger = create_autospec(Eiger, instance=True)
+    mock_eiger.status = mock_status
+    mock_eiger.settings = mock_settings
+    return mock_eiger
+
+
+@pytest.fixture
+def raise_interrupt():
+    async def raise_interrupt():
+        return False
+
+    return Mock(raise_interrupt)
+
+
+@pytest.fixture
+def eiger_adapter(mock_eiger):
+    return EigerAdapter(mock_eiger, raise_interrupt, host="localhost", port=8081)
+
+
+def test_eiger_adapter_contructor():
+    EigerAdapter(mock_eiger, raise_interrupt, host="localhost", port=8081)
+
+
+# @pytest.fixture
+# async def mock_client() -> Mock:
+#     app = web.Application()
+#     client = await aiohttp_client(app)
+#     return client
+
+
+DETECTOR_API = "http://localhost:8081/detector/api/1.8"
+
+
+@pytest.fixture
+def mock_good_get_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "count_time"}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_bad_get_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "wrong_param"}
+
+    return mock_request
```

Comment:
Could these be condensed into one fixture with an argument?

Reply:
My preference would be for a [parameterised fixture](https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments), but can also see the merits of a fixture returning a factory - particularly in cases where the relations are awkward to express tabularly.